astrolabe/runner.py

```diff
@@ -91,29 +93,31 @@ def is_cluster_state(self, goal_state):
         cluster_info = self.cluster_url.get().data
         return cluster_info.stateName.lower() == goal_state.lower()
 
-    def verify_cluster_configuration_matches(self, state):
+    def verify_cluster_configuration_matches(self, expected_configuration):
         """Verify that the cluster config is what we expect it to be (based on
         maintenance status). Raises AssertionError."""
-        state = state.lower()
-        if state not in ("initial", "final"):
-            raise AstrolabeTestCaseError(
-                "State must be either 'initial' or 'final'.")
         cluster_config = self.cluster_url.get().data
         assert_subset(
             cluster_config,
-            self.spec.maintenancePlan[state].clusterConfiguration)
+            expected_configuration.clusterConfiguration)
         process_args = self.cluster_url.processArgs.get().data
         assert_subset(
-            process_args, self.spec.maintenancePlan[state].processArgs)
+            process_args, expected_configuration.processArgs)
 
-    def initialize(self):
+    def initialize(self, no_create=False):
         """
         Initialize a cluster with the configuration required by the test
         specification.
         """
+        
+        self.start_time = _time.time()
+        
+        if no_create:
+            return
```

Comment:
Shouldn't you at least check that a cluster of the expected name exists before returning? Also, I think you should emit a `LOGGER.info` event before returning.

Reply:
You are right that the diagnostics would be better if the name was verified. Thinking about this some more, I'll make --no-create create the cluster if it doesn't exist which should make this option more usable without imposing additional runtime cost.